file-attribute-caching/src/main/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemProvider.kt

```diff
@@ -44,7 +44,8 @@ internal class AttributeCachingFileSystemProvider : FileSystemProvider() {
     private val fileSystems = mutableMapOf<URI, FileSystem>()
     private val isWindowsOS by lazy {
         // Adapted from org/junit/jupiter/api/condition/OS.java to look up the operating system name
-        System.getProperty("os.name").lowercase(Locale.ENGLISH).contains("win")
+        // lowercase() defaults to Locale.ROOT in kotlin, but toLowerCase() in java does not, BE AWARE OF THIS.
```

Comment:
Still wondering who this comment is _for_.

We're not migrating this code to Java, readers of this library can suss this information for themselves, and we've already handled the safe operation of this for them. What is there for us to warn them about?

Reply:
we can agree to disagree here, I am leaving the warning in as you literally have to dig into the code to see whats happening behind the scenes in kotlin vs java. I shouldn't have to do that to understand whats happening at a glance.